### PR TITLE
VBM-111: Forgot password feature

### PR DESCRIPTION
**Pull request by mazurGit**
https://user-images.githubusercontent.com/69375560/188074219-a4c80957-a4eb-45bf-beaa-623b94397199.mp4

### CODE REVIEW

File: packages/mobile/src/components/input/input.tsx [hunk 1]
```diff
@@ -69,7 +71,7 @@ const Input = <T extends FormControlValues>({
           onChangeText={field.onChange}
           onBlur={field.onBlur}
           placeholderTextColor={colors.placeholder}
-          secureTextEntry={secured}
+          secureTextEntry={secureTextEntry || secured}
           selectTextOnFocus={field.name === 'password' ? false : true}
```
Comment (seal323) [inline packages/mobile/src/components/input/input.tsx:75]:
```suggestion
          selectTextOnFocus={field.name !== 'password'}
```

File: packages/mobile/src/screens/auth/components/sign-in-form/sign-in-form.tsx [hunk 2]
```diff
@@ -35,6 +47,11 @@ const SignInForm: FC<Props> = ({ onSubmit }) => {
         contentContainerStyle={globalStyles.mt5}
         isSecure={true}
       />
+      <View style={globalStyles.mt5}>
+        <ButtonText alignSelf="flex-end" onPress={navigateResetPassword}>
+          Forgot password
```
Comment (seal323) [inline packages/mobile/src/screens/auth/components/sign-in-form/sign-in-form.tsx:52]: localisation?

File: packages/mobile/src/components/button-text/button-text.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import React, { FC } from 'react';
+import { TouchableOpacity, Text, TextStyle } from 'react-native';
+import { ColorPalette } from '@vse-bude/shared';
+import { globalStyles } from '~/styles/styles';
+
+type Props = {
+  onPress: () => void;
+  children: string;
+  alignSelf: 'flex-end' | 'flex-start' | 'center';
+  textStyle?: TextStyle[];
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/components/button-text/button-text.tsx:10]:
```suggestion
  contantContainerStyle?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
```

File: packages/mobile/src/components/check-box/check-box.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  };
+
+  return (
+    <RNCheckBox
+      tintColors={{
+        true: ColorPalette.YELLOW_100,
+        false: ColorPalette.YELLOW_100,
+      }}
+      value={checked}
+      onValueChange={(newValue) => onValueChange(newValue)}
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/components/check-box/check-box.tsx:25]:
```suggestion
      onValueChange={onValueChange}
```

File: packages/mobile/src/components/check-box/check-box.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  onChange: (value: boolean) => void;
+};
+
+const CheckBox: FC<Props> = ({ onChange }) => {
+  const [checked, setChecked] = useState(false);
+
+  const onValueChange = (newValue: boolean) => {
+    onChange(newValue);
+    setChecked(newValue);
+  };
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/components/check-box/check-box.tsx:16]:
```suggestion
const CheckBox: FC<Props> = ({ value, onChange }) => {
```
Actually you can just reexport default component and use it

File: packages/mobile/src/components/input/input.tsx [hunk 6]
```diff
@@ -70,7 +71,7 @@ const Input = <T extends FormControlValues>({
           onBlur={field.onBlur}
           placeholderTextColor={colors.placeholder}
           secureTextEntry={secured}
-          selectTextOnFocus={field.name === 'password' ? false : true}
+          selectTextOnFocus={field.name !== 'password'}
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/components/input/input.tsx:74]: What about repeatPassword, newPassword?

It's better to bind to `secured`

**Review by AleksandrTermenzhy — APPROVED** [review]

**Review by seal323 — APPROVED** [review]
Approve after fixes

**Pull request merged**